File: typescript/src/core/modules/auth/helpers/get-info-from-claims.ts
/**
 * Tổng hợp và lấy thông tin người dùng trong cognito claims.
 *
 * @param claims - cognito claims.
 *
 * @returns
 */
export function getInfoFromClaims(claims: any) {
  return {
    username: claims.username,
    team: claims["cognito:groups"][0],
    role: claims["custom:role"],
  };
}

File: python/src/core/modules/auth/helpers/get_info_from_claims.py
from typing import Any, Dict


def get_info_from_claims(claims: Dict[str, Any]) -> Dict[str, Any]:
    """
    Tổng hợp và lấy thông tin người dùng trong cognito claims.

    Args:
        claims (Dict[str, Any]): cognito claims.
    Return:
        dict: chứa username, team, role.
    """
    return {
        "username": claims.get("username"),
        "team": claims.get("cognito:groups", [None])[0],
        "role": claims.get("custom:role"),
    }
